python/lsst/meas/astrom/astrometry.py

```diff
@@ -215,25 +218,34 @@ def solve(self, sourceCat, bbox, initWcs, filterName=None, calib=None, exposure=
                 res = tryRes
                 break
 
-            self.log.logdebug(
+            self.log.info(
                 "Fit WCS iter %s: %s matches; median scatter = %g arcsec" % \
                     (i, len(tryRes.matches), tryRes.scatterOnSky.asArcseconds()),
             )
 
-            if res is not None and not self.config.forceKnownWcs:
-                if len(tryRes.matches) < len(res.matches):
-                    self.log.info(
-                        "Fit WCS: use iter %s because it had more matches than the next iter: %s vs. %s" % \
-                        (i-1, len(res.matches), len(tryRes.matches)))
-                    break
-                if len(tryRes.matches) == len(res.matches) and tryRes.scatterOnSky >= res.scatterOnSky:
-                    self.log.info(
-            "Fit WCS: use iter %s because it had less scatter than the next iter: %g vs. %g arcsec" % \
-                        (i-1, res.scatterOnSky.asArcseconds(), tryRes.scatterOnSky.asArcseconds()))
-                    break
+#            if res is not None and not self.config.forceKnownWcs:
+#                if len(tryRes.matches) < len(res.matches):
+#                    self.log.info(
+#                        "Fit WCS: use iter %s because it had more matches than the next iter: %s vs. %s" % \
+#                        (i-1, len(res.matches), len(tryRes.matches)))
+#                    break
+#                if len(tryRes.matches) == len(res.matches) and tryRes.scatterOnSky >= res.scatterOnSky:
+#                    self.log.info(
+#            "Fit WCS: use iter %s because it had less scatter than the next iter: %g vs. %g arcsec" % \
+#                        (i-1, res.scatterOnSky.asArcseconds(), tryRes.scatterOnSky.asArcseconds()))
+#                   break
 
             res = tryRes
             wcs = res.wcs
+            
+            # Should update the maximum distance between source and reference here, in order to reject outliers 
+            matches = res.matches
+            d = []
+
+            for match in matches :
+                d.append(match.distance)
+            print("***** distance ", afwGeom.radToArcsec(numpy.mean(d)), afwGeom.radToArcsec(numpy.std(d)))
+            minDist = afwGeom.radToArcsec(numpy.mean(d))+2.0* afwGeom.radToArcsec(numpy.std(d))
```

Comment:
We simply define the new maximum distance between source and reference as average distance + 2*sigma. This can be a tunable parameter